### ISSUE TITLE
Find() with filter works but find({}) to get all documents fails

### ISSUE THREAD

**Issue by Bryson14** [mongodb/mongo-rust-driver#1066]
## Versions/Environment
1. What version of Rust are you using? 1.77
2. What operating system are you using? WSL2
3. What versions of the driver and its dependencies are you using? (Run 
`cargo pkgid mongodb` & `cargo pkgid bson`) 2.8.2 & 2.9.2
4. What version of MongoDB are you using? (Check with the MongoDB shell using `db.version()`)  7
5. What is your MongoDB topology (standalone, replica set, sharded cluster, serverless)? standalone

## Describe the bug

I have migrated some code from dynamodb to mongodb. I have almost all of my previous routes tested and working except the list_all() functions. 

This is an example of my code:

```
pub async fn process_find<T>(mut cursor: Cursor<T>) -> Result<Vec<T>, JsonResponse>
where
	T: DeserializeOwned + Debug + Serialize,
{
	// TODO consider enabling streaming from the lambda function to reduce need for large items vec
	let mut items: Vec<T> = Vec::new();
	loop {
		match cursor.try_next().await {
			Ok(Some(result)) => items.push(result),
			Ok(None) => break, // End of cursor
			Err(e) => {
				// Handle error
				let msg = ErrorResponse {
					error_message: &format!("Failed to iterate cursor: {}", e),
				};
				error!("Failed to iterate over cursor: {:>}", e);
				return Err((StatusCode::INTERNAL_SERVER_ERROR, Json(msg.to_value())));
			}
		}
	}
	Ok(items)
}

pub async fn list(
	Extension(db): Extension<Database>,
	Query(params): Query<PaginationRequest>,
) -> WebResult {
	let db = db.clone();
	let collection = get_messages_collection(db);
	let options = FindOptions::builder()
		.limit(params.limit)
		.skip(params.skip)
		.build();
	let res = collection.find(doc! {}, options).await;
	match res {
		Ok(cursor) => process_find_and_return::<Message>(cursor).await,
		Err(e) => {
			error!("failed to run find on message collection. Error: {:?}", e);
			let msg = ErrorResponse {
				error_message: "Failed to query collection",
			};
			Err((StatusCode::INTERNAL_SERVER_ERROR, Json(msg.to_value())))
		}
	}
}


```

Error

```
 Failed to interate over cursor: Kind: invalid type: string "{\"system_prompt\":{\"S\":\"You're an experienced Software architect and Software engineer. You understand software engineering for Healthcare\\n                technologies.You have deep and broad experience in architecting, developing and testing applications in various programming\\n                languages such as Java, Springboot, Javascript,NodeJS,React,Angular C, C++, Python, Rust, GoLang,Terraform,CloudFormation,JSON,HTML,CSS,SQL,NoSQL, etc.\\n                You have deep knowledge of data structures, algorithms, concurrency, software best practices, databases, software testing , REST, networking, DevOps, Software Security, Software reliability, AI, ML and AWS, Azure, Google cloud technologies.\\n                You don't provide theoretic or generic information.  If you dont know the answer ask clarifying questions.\\n                {content}\\n               \"}}", expected a map, labels:
 ```

**To Reproduce**
1. First, do this.
2. Then do this.
3. After doing that, do this.
4. And then, finally, do this.
5. Bug occurs.

**Comment by Bryson14** [comment]
Oh I think I realized the issue. When migrating from dynamoDB to MongoDB, the nested json strings that are native to Dynamodb get brought over as strings. You can even see the weird strings on your migration docs (https://www.mongodb.com/blog/post/migrating-your-data-from-dynamodb-to-mongodb-atlas)
See this [picture here](Now that our documents are uploaded, we can log into [Atlas with MongoDB Compass](https://www.mongodb.com/blog/post/your-mongodb-atlas-toolkit-logging-into-mongodb-atlas-with-compass) and review our data:) that has the string `info: "{ actors: { 'S': "something } ....` These structures are read into rust using the [dynamo-serde](https://docs.rs/serde_dynamo/latest/serde_dynamo/) crate and the different maps, arrays can deserialized into the rust struct, similar to how bson does it.


TLDR, MongoDB's normal migration from DynamoDB cannot deal with maps, arrays, and other types that can exist in dynamoDB, so now I have to migrate that stuff.
